enterprise/api/v1/serializers.py

```diff
@@ -191,37 +244,35 @@ def save(self):  # pylint: disable=arguments-differ
         ecu.save()
 
 
-class EnterpriseCustomerUserEntitlementSerializer(serializers.Serializer):
+class EnterpriseCustomerUserEntitlementSerializer(ImmutableStateSerializer):
     """
     Serializer for the entitlements of EnterpriseCustomerUser.
 
     This Serializer is for read only endpoint of enterprise learner's entitlements
     It will ignore any state changing requests like POST, PUT and PATCH.
     """
+
     entitlements = serializers.ListField(
         child=serializers.DictField()
     )
 
-    def create(self, validated_data):
-        """
-        Do not perform any operations for state changing requests.
-        """
-        pass
-
-    def update(self, instance, validated_data):
-        """
-        Do not perform any operations for state changing requests.
-        """
-        pass
     user = UserSerializer(read_only=True)
     enterprise_customer = EnterpriseCustomerSerializer(read_only=True)
     data_sharing_consent = UserDataSharingConsentAuditSerializer(many=True, read_only=True)
 
 
-class EnterpriseCourseCatalogReadOnlySerializer(serializers.Serializer):
+class EnterpriseCustomerCatalogApiPaginatedResponseReadOnlySerializer(ResponsePaginationSerializer):
```

Comment:
I would name it with `API` but, since I think `Api` tends to be the preferred style and any renaming done in this file would be pretty easy to fix across the codebase (since it's hardly used), I'll change the other one rather than this one; since you know I'll be doing it now, the review won't come as a surprise -- is that good?